File: widget/src/main/java/com/smasher/widget/basic/BasicActivity.java
package com.smasher.widget.basic;

import android.content.Context;
import android.graphics.Color;
import android.graphics.drawable.ColorDrawable;
import android.os.Bundle;
import android.util.Log;
import android.view.View;
import android.view.animation.AccelerateInterpolator;
import android.view.animation.DecelerateInterpolator;
import android.widget.LinearLayout;

import androidx.appcompat.app.AppCompatActivity;
import androidx.appcompat.widget.Toolbar;
import androidx.coordinatorlayout.widget.CoordinatorLayout;
import androidx.fragment.app.FragmentManager;
import androidx.viewpager.widget.ViewPager;

import com.google.android.material.appbar.AppBarLayout;
import com.google.android.material.floatingactionbutton.FloatingActionButton;
import com.google.android.material.snackbar.Snackbar;
import com.smasher.oa.core.utils.StatusBarUtil;
import com.smasher.widget.R;

import net.lucode.hackware.magicindicator.MagicIndicator;
import net.lucode.hackware.magicindicator.ViewPagerHelper;
import net.lucode.hackware.magicindicator.buildins.UIUtil;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.CommonNavigator;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.CommonNavigatorAdapter;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerIndicator;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.abs.IPagerTitleView;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.indicators.LinePagerIndicator;
import net.lucode.hackware.magicindicator.buildins.commonnavigator.titles.SimplePagerTitleView;

import java.util.ArrayList;


/**
 * CoordinatorLayout等一系列的使用
 *
 * @author matao
 */
public class BasicActivity extends AppCompatActivity {

    private static final String TAG = "BasicActivity";
    Toolbar toolbar;
    FloatingActionButton fab;
    AppBarLayout appBarLayout;
    CoordinatorLayout coordinatorLayout;
    MagicIndicator magicIndicator;
    ViewPager viewPager;

    FragmentManager mFragmentManager;
    FragmentStateAdapter mFragmentStateAdapter;

    ArrayList<MFragment> mArrayList = new ArrayList<>();

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_basic);
        StatusBarUtil.setTranslucent(this, 200);

        Log.d(TAG, "onCreate: ");
        initView();
        initListener();
        initViewPager();
        initMagicIndicator();
    }

    private void initListener() {
        fab.setOnClickListener(mOnClickListener);

    }

    private void initView() {
        toolbar = findViewById(R.id.toolbar);
        fab = findViewById(R.id.fab);
        appBarLayout = findViewById(R.id.appBarLayout);
        coordinatorLayout = findViewById(R.id.coordinatorLayout);
        magicIndicator = findViewById(R.id.magicIndicator);
        viewPager = findViewById(R.id.viewPager);
        setSupportActionBar(toolbar);
        mFragmentManager = getSupportFragmentManager();
    }


    private void initViewPager() {

        try {
            mFragmentStateAdapter = new FragmentStateAdapter(getSupportFragmentManager(), mArrayList);
            viewPager.setAdapter(mFragmentStateAdapter);
            viewPager.setOffscreenPageLimit(4);
        } catch (Exception e) {
            e.printStackTrace();
        }
    }

    private void initMagicIndicator() {
        magicIndicator.setBackgroundColor(Color.parseColor("#00000000"));

        CommonNavigator commonNavigator = new CommonNavigator(this);
        commonNavigator.setAdapter(new CommonNavigatorAdapter() {
            @Override
            public int getCount() {
                return 4;
            }

            @Override
            public IPagerTitleView getTitleView(Context context, int index) {
                SimplePagerTitleView simplePagerTitleView = new ColorFlipPagerTitleView(context);
                simplePagerTitleView.setText("Title" + index);
                simplePagerTitleView.setNormalColor(Color.parseColor("#99fdda00"));
                simplePagerTitleView.setSelectedColor(Color.parseColor("#fdda00"));
                simplePagerTitleView.setOnClickListener(v -> viewPager.setCurrentItem(index));
                return simplePagerTitleView;
            }

            @Override
            public IPagerIndicator getIndicator(Context context) {
                LinePagerIndicator indicator = new LinePagerIndicator(context);
                indicator.setYOffset(10);
                indicator.setMode(LinePagerIndicator.MODE_EXACTLY);
                indicator.setLineHeight(UIUtil.dip2px(context, 6));
                indicator.setLineWidth(UIUtil.dip2px(context, 30));
                //indicator.setRoundRadius(UIUtil.dip2px(context, 3));
                indicator.setStartInterpolator(new AccelerateInterpolator());
                indicator.setEndInterpolator(new DecelerateInterpolator(2.0f));
                indicator.setColors(Color.parseColor("#00c853"));
                return indicator;
            }
        });
        magicIndicator.setNavigator(commonNavigator);
        LinearLayout linearLayout = commonNavigator.getTitleContainer();
        linearLayout.setShowDividers(LinearLayout.SHOW_DIVIDER_MIDDLE | LinearLayout.SHOW_DIVIDER_BEGINNING | LinearLayout.SHOW_DIVIDER_END);
        linearLayout.setDividerPadding(UIUtil.dip2px(this, 35));
        linearLayout.setDividerDrawable(new ColorDrawable(Color.parseColor("#00000000")) {

            @Override
            public int getIntrinsicWidth() {
                return UIUtil.dip2px(BasicActivity.this, 25);
            }
        });


        ViewPagerHelper.bind(magicIndicator, viewPager);
    }


    View.OnClickListener mOnClickListener = view -> {
        int i = view.getId();
        if (i == R.id.fab) {
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show();
        }
    };
}

File: rejuvenation/src/main/java/com/smasher/rejuvenation/activity/BasicActivity.kt
package com.smasher.rejuvenation.activity

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.view.LayoutInflater
import android.view.View
import android.widget.Button
import com.google.android.material.snackbar.Snackbar
import com.smasher.aidl.activity.local.LocalActivity
import com.smasher.aidl.activity.remote.ui.login.LoginActivity
import com.smasher.draw.activity.DrawableActivity
import com.smasher.media.activity.TestActivity
import com.smasher.oa.core.utils.StatusBarUtil
import com.smasher.rejuvenation.R
import com.smasher.widget.activity.AlarmActivity
import com.smasher.widget.activity.WidgetActivity
import com.smasher.widget.base.BaseActivity
import kotlinx.android.synthetic.main.activity_basics.*

class BasicActivity : BaseActivity(), View.OnClickListener {


    private var mButton1: Button? = null
    private var mButton2: Button? = null
    private var mButton3: Button? = null
    private var mButton4: Button? = null
    private var mButton5: Button? = null
    private var mButton6: Button? = null
    private var mButton7: Button? = null
    private var mButton8: Button? = null


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        StatusBarUtil.setTranslucent(this)
        toolbar.title = "BasicActivity"
        setSupportActionBar(toolbar)
        initListener()
    }


    override fun getRootView(): View {
        return LayoutInflater.from(this).inflate(R.layout.activity_basics, null)
    }

    override fun initView() {
        mButton1 = findViewById(R.id.button1)
        mButton2 = findViewById(R.id.button2)
        mButton3 = findViewById(R.id.button3)
        mButton4 = findViewById(R.id.button4)
        mButton5 = findViewById(R.id.button5)
        mButton6 = findViewById(R.id.button6)
        mButton7 = findViewById(R.id.button7)
        mButton8 = findViewById(R.id.button8)
    }


    override fun setFunctionsForFragment(tag: String?) {

    }

    override fun initData() {

    }


    private fun initListener() {
        mButton1!!.setOnClickListener(this)
        mButton2!!.setOnClickListener(this)
        mButton3!!.setOnClickListener(this)
        mButton4!!.setOnClickListener(this)
        mButton5!!.setOnClickListener(this)
        mButton6!!.setOnClickListener(this)
        mButton7!!.setOnClickListener(this)
        mButton8!!.setOnClickListener(this)
        fab.setOnClickListener { view ->
            Snackbar.make(view, "Replace with your own action", Snackbar.LENGTH_LONG)
                    .setAction("Action", null).show()
        }
    }


    override fun onClick(v: View?) {
        when (v!!.id) {
            R.id.button1 -> gotoLogin()
            R.id.button2 -> gotoAlarm()
            R.id.button3 -> gotoDrawable()
            R.id.button4 -> gotoLocal()
            R.id.button5 -> gotoWidget()
            R.id.button6 -> gotoTest()
            R.id.button7 -> Log.d("TAG", "button")
            R.id.button8 -> Log.d("TAG", "button")
            else -> Log.d("TAG", "button")
        }
    }

    private fun gotoLogin() {
        val intent = Intent()
        intent.setClass(this, LoginActivity::class.java)
        startActivity(intent)
    }

    private fun gotoAlarm() {
        val intent = Intent()
        intent.setClass(this, AlarmActivity::class.java)
        startActivity(intent)
    }

    private fun gotoDrawable() {
        val intent = Intent()
        intent.setClass(this, DrawableActivity::class.java)
        startActivity(intent)
    }

    private fun gotoLocal() {
        val intent = Intent()
        intent.setClass(this, LocalActivity::class.java)
        startActivity(intent)
    }

    private fun gotoWidget() {
        val intent = Intent()
        intent.setClass(this, WidgetActivity::class.java)
        startActivity(intent)
    }

    private fun gotoTest() {
        val intent = Intent()
        intent.setClass(this, TestActivity::class.java)
        startActivity(intent)
    }

}
